src/Form/FormControlFeedback.jsx

```diff
@@ -0,0 +1,96 @@
+import React, { useEffect, useState } from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+import { useFormGroupContext } from './FormGroupContext';
+import { Icon } from '..';
+import {
+  Check, Close, Cancel, CheckCircle, RadioButtonUnchecked, WarningFilled,
+} from '../../icons';
+
+const FEEDBACK_TYPES = {
+  VALID: 'valid',
+  INVALID: 'invalid',
+  WARNING: 'warning',
+  CRITERIA_EMPTY: 'criteria-empty',
+  CRITERIA_VALID: 'criteria-valid',
+  CRITERIA_INVALID: 'criteria-invalid',
+};
+
+const FEEDBACK_ICONS = {
+  [FEEDBACK_TYPES.VALID]: Check,
+  [FEEDBACK_TYPES.INVALID]: Close,
+  [FEEDBACK_TYPES.WARNING]: WarningFilled,
+  [FEEDBACK_TYPES.CRITERIA_EMPTY]: RadioButtonUnchecked,
+  [FEEDBACK_TYPES.CRITERIA_VALID]: CheckCircle,
+  [FEEDBACK_TYPES.CRITERIA_INVALID]: Cancel,
+};
+
+const FeedbackIcon = ({ type, customIcon }) => {
+  if (customIcon) {
+    return customIcon;
+  }
+
+  const typeIcon = FEEDBACK_ICONS[type];
+  if (typeIcon) {
+    return <Icon src={typeIcon} />;
+  }
+
+  return null;
+};
+
+FeedbackIcon.propTypes = {
+  type: PropTypes.oneOf(Object.values(FEEDBACK_TYPES)),
+  customIcon: PropTypes.node,
+};
+
+FeedbackIcon.defaultProps = {
+  type: undefined,
+  customIcon: undefined,
+};
+
+const FormControlFeedback = ({
+  children, type, icon, className, muted, ...props
+}) => {
+  const { controlId, getNewDescriptorId } = useFormGroupContext();
+  const [id, setId] = useState();
+  useEffect(() => setId(getNewDescriptorId()), [controlId]);
```

Comment:
Talk to me about what controlId is doing here and why we need to generate a new id when it changes?

Reply:
While not strictly necessary, since the controlId is used as the root of the generated description id, I figured we'd regenerated it. Maybe not a great choice.